pkg/agent/datamodel/const.go

```diff
@@ -132,19 +140,71 @@ const (
 	EnableWinDSR          = "EnableWinDSR"
 )
 
-const (
-	Nvidia470CudaDriverVersion = "cuda-470.82.01"
-	Nvidia550CudaDriverVersion = "550.90.12"
-	Nvidia535GridDriverVersion = "535.161.08"
-)
+const Nvidia470CudaDriverVersion = "cuda-470.82.01"
 
-// These SHAs will change once we update aks-gpu images in aks-gpu repository. We do that fairly rarely at this time.
-// So for now these will be kept here like this.
-const (
-	AKSGPUCudaVersionSuffix = "20241021235610"
-	AKSGPUGridVersionSuffix = "20241021235607"
+//nolint:gochecknoglobals
+var (
+	NvidiaCudaDriverVersion string
+	NvidiaGridDriverVersion string
+	AKSGPUCudaVersionSuffix string
+	AKSGPUGridVersionSuffix string
 )
 
+type gpuVersion struct {
+	RenovateTag   string `json:"renovateTag"`
+	LatestVersion string `json:"latestVersion"`
+}
+
+type gpuContainerImage struct {
+	DownloadURL string     `json:"downloadURL"`
+	GPUVersion  gpuVersion `json:"gpuVersion"`
+}
+
+type componentsConfig struct {
+	GPUContainerImages []gpuContainerImage `json:"GPUContainerImages"`
+}
+
+func LoadConfig() error {
+	// Read the embedded components.json file
+	data, err := parts.Templates.ReadFile("linux/cloud-init/artifacts/components.json")
+	if err != nil {
+		return fmt.Errorf("failed to read components.json: %w", err)
+	}
+
+	var config componentsConfig
+	if err := json.Unmarshal(data, &config); err != nil {
+		return fmt.Errorf("failed to unmarshal components.json: %w", err)
+	}
+
+	const driverIndex = 0
+	const suffixIndex = 1
+	const expectedLength = 2
+
+	for _, image := range config.GPUContainerImages {
+		parts := strings.Split(image.GPUVersion.LatestVersion, "-")
+		if len(parts) != expectedLength {
+			continue
+		}
+		version, suffix := parts[driverIndex], parts[suffixIndex]
+
+		if strings.Contains(image.DownloadURL, "aks-gpu-cuda") {
+			NvidiaCudaDriverVersion = version
+			AKSGPUCudaVersionSuffix = suffix
+		} else if strings.Contains(image.DownloadURL, "aks-gpu-grid") {
+			NvidiaGridDriverVersion = version
+			AKSGPUGridVersionSuffix = suffix
+		}
+	}
+	return nil
+}
+
+//nolint:gochecknoinits
+func init() {
+	if err := LoadConfig(); err != nil {
+		panic(fmt.Sprintf("Failed to load configuration: %v", err))
+	}
+}
+
```

Comment:
maybe move all this to another file `gpu_components.go`? `consts.go` probably not the best place

Reply:
fair, moved it